client/ayon_usd/standalone/Usd/Pinning/pinning_file_helper.py

```diff
@@ -0,0 +1,235 @@
+import json
+import os
+import re
+from typing import Dict, List
+from pxr import UsdShade, Ar, Sdf
+from urllib.parse import urlparse
+
+
+def is_uri(path):
+    parsed = urlparse(path)
+    return parsed.scheme
+
+
+# we do the assumption that the environment sets up the correct Default resolver (in AYON case this should be the AyonUsdResolver)
+resolver = Ar.GetResolver()
+
+
+def remove_root_from_dependecy_info(
+    dependecy_info: Dict[str, str], root_info: Dict[str, str]
+) -> Dict[str, str]:
+    """function that re_roots a path dict so that a Ayon_Usd_Resolver can use its own system for root replace to resolve a given path
+
+    Args:
+        dependecy_info:
+        root_info: a flat dict containing str: root identifier {root[work]}, str: associated root path /path/to/root
+
+    Returns: the input dependecy_info dict with replaced root_info to obtain root-less paths,
```

Comment:
```suggestion
    Returns:
        The input dependency_info dict with replaced root_info to obtain root-less paths.
```

It's unclear from the docstring what "dependency info" is and maybe even wat "root info" is. As mentioned before, an example may help in the docstring.

Reply:
done in separate commit. 
wont include code example but i pointed to all the other functions that will give you the needed data 
eg Resolve Uris, Get Project Overwrites, etc.